### PR TITLE
WIP: Use the base class

### PR DESCRIPTION
**Pull request by diasdavid**
> This makes libp2p bundles so simple!

ref: https://github.com/libp2p/js-libp2p/pull/43

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,202 +1,33 @@
...
-
-  this.handle = (protocol, handlerFunc, matchFunc) => {
-    return this.swarm.handle(protocol, handlerFunc, matchFunc)
-  }
-
-  this.unhandle = (protocol) => {
-    return this.swarm.unhandle(protocol)
+const libp2p = require('libp2p')
+
+class Node extends libp2p.Node {
```
Comment (dignifiedquire) [inline src/index.js]: Sometimes inheritance is just awesome :)

**Comment by diasdavid** [comment]
It works \o/ So happy about this refactoring, makes things so much nicer and understandable

@dignifiedquire Not sure if I should consider your comment as a positive CR, but I believe it is safe to assume that if you found an issue, you would comment right away :) Will merge once CI is green

**Comment by dignifiedquire** [comment]
@diasdavid I didn't do any real CR yet will do tomorrow

**Comment by diasdavid** [comment]
thank you :)

**Comment by diasdavid** [comment]
while you are at it :) https://github.com/ipfs/js-libp2p-ipfs-browser/pull/126

File: src/index.js [hunk 2]
```diff
@@ -1,202 +1,33 @@
...
+    }
+    super(modules, peerInfo, peerBook)
   }
+}
 
-  this.discovery = new EE()
-  this.routing = null
-  this.records = null
+module.exports = {
+  Node: Node
```
Comment (dignifiedquire) [inline src/index.js]: just `module.exports = Node`

Comment (dignifiedquire) [inline src/index.js]: not really, I would just like to write 

```js
const Node = require('libp2p-ipfs')

const myNode = new Node()
```

Comment (diasdavid) [inline src/index.js]: fine

Comment (diasdavid) [inline src/index.js]: done :)

**Pull request merged**